clippy_lints/src/mem_replace.rs

```diff
@@ -238,6 +283,13 @@ impl<'tcx> LateLintPass<'tcx> for MemReplace {
             } else if self.msrv.meets(msrvs::MEM_TAKE) && is_expr_used_or_unified(cx.tcx, expr) {
                 check_replace_with_default(cx, src, dest, expr.span);
             }
+            // Check that second argument is `Option::Some`
```

Comment:
Shouldn't that be part of the check method? To me as a reviewer, this splits the code and if I look at the check method in isolation, it's not clear what actually is checked.

Reply:
Totally. I have also reorganized the other checks so that they are more self-contained, in the first commit.